designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala

```diff
@@ -109,10 +116,69 @@ class DeploymentServiceSpec
       processValidatorByProcessingType,
       TestFactory.scenarioResolverByProcessingType,
       listener,
-      scenarioStateTimeout = scenarioStateTimeout
+      scenarioStateTimeout = scenarioStateTimeout,
+      deploymentCommentSettings
     )
   }
 
+  // TODO: temporary step - we would like to extract the validation and the comment validation tests to external validators
+
+  private val nonEmptyCommentSettings = Some(createNonEmptyDeploymentCommentSettings())
+
+  def createDeploymentServiceWithCommentSettings(): DeploymentService = {
+    createDeploymentService(deploymentCommentSettings = nonEmptyCommentSettings)
+  }
+
+  test { "should return error when trying to deploy without comment when comment is required" } {
+    val deploymentServiceWithCommentSettings = createDeploymentServiceWithCommentSettings()
+    val processName: ProcessName             = generateProcessName
+    val id                                   = prepareProcess(processName).dbioActionValues
+
+    val result = deploymentServiceWithCommentSettings.deployProcessAsync(id, None, None).failed.futureValue
+
+    result shouldBe a[ValidationError]
+    result.getMessage.trim shouldBe "Comment is required."
```

Comment:
It looks like we need extra check here, to assert that all started actions are marked as finished or closed/removed. Right now cypress tests fail because of DEPLOY action is hanging in "in progress" state. For all tests where we expect returned validation error please add something like this:
```
    eventually {
      val inProgressActions = actionRepository.getInProgressActionTypes(processId.id).dbioActionValues
      inProgressActions should have size 0
    }
```